cli/cmd/project/clone.go

```diff
@@ -0,0 +1,153 @@
+package project
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	githttp "github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/rilldata/rill/cli/cmd/env"
+	"github.com/rilldata/rill/cli/pkg/cmdutil"
+	"github.com/rilldata/rill/cli/pkg/dotrillcloud"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/spf13/cobra"
+)
+
+func CloneCmd(ch *cmdutil.Helper) *cobra.Command {
+	var path string
+
+	cloneCmd := &cobra.Command{
+		Use:   "clone [<project-name>]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Clone Project",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			name := args[0]
+			if path == "" {
+				path = name
+			}
+
+			client, err := ch.Client()
+			if err != nil {
+				return err
+			}
+
+			// get project
+			res, err := client.GetProject(cmd.Context(), &adminv1.GetProjectRequest{OrganizationName: ch.Org, Name: name})
+			if err != nil {
+				return err
+			}
+
+			if res.Project.ArchiveAssetId != "" {
+				return fmt.Errorf("project is not connected to a git repository, please redeploy the project to connect it to a git repository")
+			}
+
+			// check if dir is empty
+			empty, err := isDirAbsentOrEmpty(path)
+			if err != nil {
+				return err
+			}
+			if !empty {
+				ok, err := cmdutil.ConfirmPrompt(fmt.Sprintf("Directory %q is not empty. Do you want to overwrite it?", path), "", false)
```

Comment:
We don't know that it is a directory, it could also be a file:
```suggestion
				ok, err := cmdutil.ConfirmPrompt(fmt.Sprintf("There are files at path %q. Do you want to overwrite?", path), "", false)
```